File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        
        if(x < 0) return false;

        long long res = 0;
        long long x_ = x;
        
        while(x_ != 0){
            int temp = x_%10;
            x_ /= 10;
            res = res*10 + temp;
        }
        
        return x == res ? true:false;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        x_ = x
        reverse_x = 0
        while x_ is not 0:
            temporal = x_ % 10
            x_ = x_//10
            reverse_x = reverse_x * 10 + temporal

        if x is not reverse_x:
            return False

        return True
        